From 004a36e0e75bf31d41bb679a64e8a4e320020c6a Mon Sep 17 00:00:00 2001
From: Alexandru Gagniuc <mr.nuke.me@gmail.com>
Date: Sat, 23 Mar 2024 19:23:16 -0500
Subject: [PATCH 13/19] edma-v2: remove the obsolte u64_stats_fetch_*_irq()
 users.

Since linux 6.1, the *_irq() functions have been obsoleted, tith the
non-irq variants intended to replace them. The _irq versions were
removed in linux 6.3.

Update the edma-v2 implementation to use the new variants.

See the folluwing linux commits for more details:

 * dec5efcffad4 ("u64_stat: Remove the obsolete fetch_irq()
                 variants.")
 * 44b0c2957adc ("u64_stats: Streamline the implementation")
 * 97c4090badca ("bpf: Remove the obsolte u64_stats_fetch_*_irq()
                 users.")
 * d120d1a63b2c ("net: Remove the obsolte u64_stats_fetch_*_irq()
                 users (net).")
 * 068c38ad88cc ("net: Remove the obsolte u64_stats_fetch_*_irq()
                 users (drivers).")
 * 93cc2559d3fd ("spi: Remove the obsolte u64_stats_fetch_*_irq()
                 users.")

Signed-off-by: Alexandru Gagniuc <mr.nuke.me@gmail.com>
---
 hal/dp_ops/edma_dp/edma_v2/edma.c         |  4 ++--
 hal/dp_ops/edma_dp/edma_v2/edma_debugfs.c | 20 ++++++++++----------
 hal/dp_ops/edma_dp/edma_v2/edma_dp.c      |  8 ++++----
 3 files changed, 16 insertions(+), 16 deletions(-)

diff --git a/hal/dp_ops/edma_dp/edma_v2/edma.c b/hal/dp_ops/edma_dp/edma_v2/edma.c
index 8156671..c6b3d54 100644
--- a/hal/dp_ops/edma_dp/edma_v2/edma.c
+++ b/hal/dp_ops/edma_dp/edma_v2/edma.c
@@ -77,10 +77,10 @@ bool edma_nsm_sawf_sc_stats_read(struct nss_dp_hal_nsm_sawf_sc_stats *nsm_stats,
 
 	sawf_sc_stats = &edma_gbl_ctx.sawf_sc_stats[service_class];
 	do {
-		start = u64_stats_fetch_begin_irq(&sawf_sc_stats->syncp);
+		start = u64_stats_fetch_begin(&sawf_sc_stats->syncp);
 		nsm_stats->rx_packets = sawf_sc_stats->rx_packets;
 		nsm_stats->rx_bytes = sawf_sc_stats->rx_bytes;
-	} while (u64_stats_fetch_retry_irq(&sawf_sc_stats->syncp, start));
+	} while (u64_stats_fetch_retry(&sawf_sc_stats->syncp, start));
 
 	return true;
 }
diff --git a/hal/dp_ops/edma_dp/edma_v2/edma_debugfs.c b/hal/dp_ops/edma_dp/edma_v2/edma_debugfs.c
index 8d9051a..5208a90 100644
--- a/hal/dp_ops/edma_dp/edma_v2/edma_debugfs.c
+++ b/hal/dp_ops/edma_dp/edma_v2/edma_debugfs.c
@@ -91,10 +91,10 @@ static int edma_debugfs_rx_rings_stats_show(struct seq_file *m, void __attribute
 		rxfill_ring = &egc->rxfill_rings[i];
 		stats = &rxfill_ring->rx_fill_stats;
 		do {
-			start = u64_stats_fetch_begin_irq(&stats->syncp);
+			start = u64_stats_fetch_begin(&stats->syncp);
 			rx_fill_stats[i].alloc_failed = stats->alloc_failed;
 			rx_fill_stats[i].page_alloc_failed = stats->page_alloc_failed;
-		} while (u64_stats_fetch_retry_irq(&stats->syncp, start));
+		} while (u64_stats_fetch_retry(&stats->syncp, start));
 	}
 
 	/*
@@ -107,11 +107,11 @@ static int edma_debugfs_rx_rings_stats_show(struct seq_file *m, void __attribute
 		rxdesc_ring = &egc->rxdesc_rings[i];
 		stats = &rxdesc_ring->rx_desc_stats;
 		do {
-			start = u64_stats_fetch_begin_irq(&stats->syncp);
+			start = u64_stats_fetch_begin(&stats->syncp);
 			rx_desc_stats[i].src_port_inval = stats->src_port_inval;
 			rx_desc_stats[i].src_port_inval_type = stats->src_port_inval_type;
 			rx_desc_stats[i].src_port_inval_netdev = stats->src_port_inval_netdev;
-		} while (u64_stats_fetch_retry_irq(&stats->syncp, start));
+		} while (u64_stats_fetch_retry(&stats->syncp, start));
 	}
 
 	edma_debugfs_print_banner(m, EDMA_RX_RING_STATS_NODE_NAME);
@@ -181,10 +181,10 @@ static int edma_debugfs_tx_rings_stats_show(struct seq_file *m, void __attribute
 		txdesc_ring = &egc->txdesc_rings[i];
 		stats = &txdesc_ring->tx_desc_stats;
 		do {
-			start = u64_stats_fetch_begin_irq(&stats->syncp);
+			start = u64_stats_fetch_begin(&stats->syncp);
 			tx_desc_stats[i].no_desc_avail = stats->no_desc_avail;
 			tx_desc_stats[i].tso_max_seg_exceed = stats->tso_max_seg_exceed;
-		} while (u64_stats_fetch_retry_irq(&stats->syncp, start));
+		} while (u64_stats_fetch_retry(&stats->syncp, start));
 	}
 
 	/*
@@ -197,12 +197,12 @@ static int edma_debugfs_tx_rings_stats_show(struct seq_file *m, void __attribute
 		txcmpl_ring = &egc->txcmpl_rings[i];
 		stats = &txcmpl_ring->tx_cmpl_stats;
 		do {
-			start = u64_stats_fetch_begin_irq(&stats->syncp);
+			start = u64_stats_fetch_begin(&stats->syncp);
 			tx_cmpl_stats[i].invalid_buffer = stats->invalid_buffer;
 			tx_cmpl_stats[i].errors = stats->errors;
 			tx_cmpl_stats[i].desc_with_more_bit = stats->desc_with_more_bit;
 			tx_cmpl_stats[i].no_pending_desc = stats->no_pending_desc;
-		} while (u64_stats_fetch_retry_irq(&stats->syncp, start));
+		} while (u64_stats_fetch_retry(&stats->syncp, start));
 	}
 
 	edma_debugfs_print_banner(m, EDMA_TX_RING_STATS_NODE_NAME);
@@ -258,7 +258,7 @@ static int edma_debugfs_misc_stats_show(struct seq_file *m, void __attribute__((
 	for_each_possible_cpu(cpu) {
 		pcpu_misc_stats = per_cpu_ptr(edma_gbl_ctx.misc_stats, cpu);
 		do {
-			start = u64_stats_fetch_begin_irq(&pcpu_misc_stats->syncp);
+			start = u64_stats_fetch_begin(&pcpu_misc_stats->syncp);
 			misc_stats->edma_misc_axi_read_err +=
 				pcpu_misc_stats->edma_misc_axi_read_err;
 			misc_stats->edma_misc_axi_write_err +=
@@ -275,7 +275,7 @@ static int edma_debugfs_misc_stats_show(struct seq_file *m, void __attribute__((
 				pcpu_misc_stats->edma_misc_tx_timeout;
 			misc_stats->edma_misc_tx_cmpl_buf_full +=
 				pcpu_misc_stats->edma_misc_tx_cmpl_buf_full;
-		} while (u64_stats_fetch_retry_irq(&pcpu_misc_stats->syncp, start));
+		} while (u64_stats_fetch_retry(&pcpu_misc_stats->syncp, start));
 	}
 
 	edma_debugfs_print_banner(m, EDMA_MISC_STATS_NODE_NAME);
diff --git a/hal/dp_ops/edma_dp/edma_v2/edma_dp.c b/hal/dp_ops/edma_dp/edma_v2/edma_dp.c
index f094746..ee9a44a 100644
--- a/hal/dp_ops/edma_dp/edma_v2/edma_dp.c
+++ b/hal/dp_ops/edma_dp/edma_v2/edma_dp.c
@@ -264,9 +264,9 @@ static void edma_dp_get_ndo_stats(struct nss_dp_data_plane_ctx *dpc,
 		pcpu_rx_stats = per_cpu_ptr(dp_info->pcpu_stats.rx_stats, i);
 
 		do {
-			start = u64_stats_fetch_begin_irq(&pcpu_rx_stats->syncp);
+			start = u64_stats_fetch_begin(&pcpu_rx_stats->syncp);
 			memcpy(&rxp, pcpu_rx_stats, sizeof(*pcpu_rx_stats));
-		} while (u64_stats_fetch_retry_irq(&pcpu_rx_stats->syncp, start));
+		} while (u64_stats_fetch_retry(&pcpu_rx_stats->syncp, start));
 
 		stats->stats.rx_packets += rxp.rx_pkts;
 		stats->stats.rx_bytes += rxp.rx_bytes;
@@ -278,9 +278,9 @@ static void edma_dp_get_ndo_stats(struct nss_dp_data_plane_ctx *dpc,
 		pcpu_tx_stats = per_cpu_ptr(dp_info->pcpu_stats.tx_stats, i);
 
 		do {
-			start = u64_stats_fetch_begin_irq(&pcpu_tx_stats->syncp);
+			start = u64_stats_fetch_begin(&pcpu_tx_stats->syncp);
 			memcpy(&txp, pcpu_tx_stats, sizeof(*pcpu_tx_stats));
-		} while (u64_stats_fetch_retry_irq(&pcpu_tx_stats->syncp, start));
+		} while (u64_stats_fetch_retry(&pcpu_tx_stats->syncp, start));
 
 		stats->stats.tx_packets += txp.tx_pkts;
 		stats->stats.tx_bytes += txp.tx_bytes;
-- 
2.40.1

